### PR TITLE
Automated cherry pick of #97977: fix the deadlock in priority and fairness config controller

### PR DESCRIPTION
**Pull request by tkashem**
Cherry pick of #97977 on release-1.18.

#97977: fix the deadlock in priority and fairness config controller

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go [hunk 1]
```diff
@@ -221,14 +222,28 @@ func (cfgCtl *configController) initializeConfigController(informerFactory kubei
 		}})
 }
 
+// used from the unit tests only.
+func (cfgCtl *configController) getPriorityLevelState(plName string) *priorityLevelState {
```
Comment (MikeSpreitzer) [inline staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go:226]: See https://github.com/kubernetes/kubernetes/pull/97977#discussion_r556268442

**Review by MikeSpreitzer — APPROVED** [review]
/lgtm

**Review by hasheddan — APPROVED** [review]
/lgtm

**Pull request merged**